### ISSUE TITLE
expr.equals(other) scales badly

### ISSUE THREAD

**Issue by tomlarkworthy** [josdejong/mathjs#2300]
I traced a performance issue to the expressions equals method, replacing with lodash's deep equals solved it

```
equalsIsSlow = {
  const n = 20;
  const termsA = Array.from({ length: n })
    .map((_, i) => `x${i}`)
    .join(" + ");

  const termsB = Array.from({ length: n })
    .map((_, i) => `x${i}`)
    .join(" + ");

  const a = math.parse(termsA);
  const b = math.parse(termsB);

  // return a.equals(b) // Very slow, uses home rolled deep equals https://github.com/josdejong/mathjs/blob/c6cbf5538915c8964b70a1af086f47c2c0be33df/src/expression/node/Node.js#L221
  return _.isEqual(a, b); // much faster
}
```

**Comment by m93a** [comment]
Hey Tom,
thank you for reporting this! Using `lodash/isEqual` for deep equality checks sounds like a reasonable thing to do.

PRs welcome :)

**Comment by josdejong** [comment]
Fixed in `mathjs@9.4.5` via #2301, thanks @tomlarkworthy.
